pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -61,18 +66,59 @@ type Manager interface {
 
 	// RemoveOrphanedPods removes the stored state for any pods not included in the set of remaining pods.
 	RemoveOrphanedPods(remainingPods sets.Set[types.UID])
+
+	// Run starts the allocation manager. This is currently only used to handle periodic retry of
+	// pending resizes.
+	Run()
+
+	// PushPendingResize queues a pod with a pending resize request for later reevaluation.
+	// Returns true if the pod was already in the pending resizes list.
+	PushPendingResize(pod *v1.Pod) bool
+
+	// RetryPendingResizes signals the allocation manager to retry all pending resize requests.
+	RetryPendingResizes()
+
+	// HandlePodResourcesResize determines if a pod resize is feasible. If so,
+	// it updates the allocation and sets the pod resize conditions accordingly.
+	HandlePodResourcesResize(pod *v1.Pod) error
```

Comment:
I didn't add unit tests to allocation_manager_test.go because these are all pretty much covered by kubelet_test.go. We can reorganize the unit tests but for ease of review (this PR is pretty large and complicated) and since the coverage _is_ there, I was thinking about leaving that as a follow-up? Will defer to you though @tallclair